### ISSUE TITLE
Frontiers in Climate: Author's Proof corrections

### ISSUE THREAD

**Issue by cisaacstern** [pangeo-forge/roadmap#43]
All co-authors tagged in this post should now be able to download `Authors_Proof.pdf` from the file browser of [our shared Overleaf project](https://www.overleaf.com/project/6103c3cb2cc6aad53b3bf81d).  The first page of that document describes the process for submitting corrections, which ultimately need to be delivered to the journal as inline comments on a single PDF. If you'd like to provide any notes, you could either:

1. Make inline comments on a copy of `Authors_Proof.pdf` 
2. List your comments by line number in a plain text document

in either case, you can send them to me via email for aggregation into a single document for submission.

A few specific questions:

- @rwegener2, is Department of Atmospheric and Ocean Science the correct departmental affiliation for you?
- @chiaral & @sharkinsspatial, the journal was not able to identify a Frontiers LOOP profile to link your names to:
    - If you have one, let me know the link;
    - If you don't have one and would like to be linked in the final paper, please [register](https://www.frontiersin.org/register) and then let me know the link;
    - If you don't have one and don't care about being linked, do nothing 😄 
- Do we want to adjust our keywords? We can have between 5 - 8; our current selection is:
    - **data, community, cloud, ARCO, NetCDF, Zarr, python**

Other points which I'm already tracking:

- Biggest problem: @alxmrs name did not make it onto the proof (in any spelling, correct or otherwise). I will address this.
- Listing 1 is broken by a column break, which is suboptimal for a code example.
- Line 1345: "... distributed across ~~on~~ large numbers of lightweight compute nodes..."
- Various bibliography questions raised by the journal in the queries section of the proof

The journal's email to me this morning requested a response within two business days, which strikes me as a bit... fast. I've requested an extension and will report back here when I hear back on that. Either way, I'll make sure all essential points (Alex's authorship, etc.) are addressed by whenever the deadline ends up being.

cc @rabernat @jhamman

**Comment by rwegener2** [comment]
Department of Atmospheric and Oceanic Science is the correct as an affiliation for me.

I can't find anything else to mention on a final skim. Thanks for leading this process, @cisaacstern.

**Comment by rabernat** [comment]
One slight change that I would love to try to sneak into the proof is to tweak our usage of the acronym ETL. 

I have learned that ETL is considered somewhat passé in the data engineering world - e.g. https://fivetran.com/blog/etl-vs-elt.

I'm not sure these semantics are that relevant to our use case, but in any case, we might consider revising this language slightly.

**Comment by cisaacstern** [comment]
> tweak our usage of the acronym ETL

@rabernat, reading the link you provided, plus other sources like [this](https://www.datanami.com/2020/01/20/eol-of-etl/), I do see now that the _Transformation_ stage of ETL is commonly understood to connote _reduction_, which we are obviously not doing. Great catch. In the context of the authors proof stage of editing, the lightest touch correction I can see is to just coin our own acronym as a drop-in replacement.

I propose **_Extraction Optimization and Storage (EOS)_**. (This, instead of _Extraction Optimization and Loading (EOL)_, because _EOL_ has a well-known meaning.) We're lucky in that we only used ETL on Page 4, AFAICT, so the relevant section of that page would be corrected as follows:


```diff
  376 3.1. Recipes: Object-Oriented Extraction,
- 377 Transformation, and Loading (ETL)
+ 377 Optimization, and Storage (EOS)
  378 Algorithms

  ...

  400                   ... In Pangeo Forge, the operations embedded
  401 in the template algorithms are, broadly speaking, those of data 
- 402 extraction, transformation, loading (ETL). First, data is extracted 
+ 402 extraction, optimization, storage (EOS). First, data is extracted 
  403 from a traditional source file server, most commonly via HTTP or
  404 FTP request; next, the source data is transformed into an ARCO
- 405 format; and finally, the data is loaded (i.e., deposited) into cloud 
+ 405 format; and finally, the data is deposited into cloud 
  406 object storage.

- 407 Within a given class of these ETL algorithms, it’s possible to
+ 407 Within a given class of these EOS algorithms, it’s possible to
  408 largely generalize the esoteric transformation logic itself, while
  409 leaving the specific attributes, such as source file location and
  410 alignment criteria, up to the recipe contributor to fill in. The
  411 completed recipe.py module containing a specific instance
- 412 of the generic ETL algorithm can then be executed in one of a
+ 412 of the generic EOS algorithm can then be executed in one of a
  413 number of ways. ...
```
Thoughts? Can we do better than _EOS_ for a drop-in fix?

**Comment by rabernat** [comment]
Let's not invent a new acronym here. Let's just call it E[T]L, to clarify that the the transformation part is optional and customizable.

**Comment by rabernat** [comment]
> I do see now that the _Transformation_ stage of ETL is commonly understood to connote _reduction_

This is a great insight. "Transformation" is what we are doing the the examples at the end of the paper. So in that context, we are using an ELT workflow, not ETL.

**Comment by rabernat** [comment]
I am happy to defer to your judgement on what is best. EOL is fine with me.

**Comment by cisaacstern** [comment]
> EOL is fine

IMHO, if we're making up an acronym anyway, then going with _EOS_ is worthwhile to avoid overlap with [End-of-life (EOL)](https://en.wikipedia.org/wiki/End-of-life_product).

FWIW, my goal is not for EOS to stick, but rather see it as a "locally scoped variable" to get us out of this specific bind.

**Comment by cisaacstern** [comment]
I've submitted our Author's Proof Corrections to the journal, addressing all of the issues discussed here, including:

- Adding @alxmrs name to author list
- Clarifying @rwegener2's affiliation
- Correcting use of "ETL" (as suggested by @rabernat) in text to "EOS" (my solution to this problem, as described above). And I realized just in time that this also required updating Figure 4 https://github.com/pangeo-forge/flow-charts/pull/6.
- So many bibliography issues!
- Correcting text wrapping issues in all three of the Listings. (The submitted manuscript was single-columned, but the Proof is two-columned. This shortened the line length for Listings, resulting in text wrapping which was not reflective of valid Python code.)

I feel like as a paper about "Good ideas in Python" our code Listings should at least be readable Python. So I've requested an additional Author Proof prior to publication, so that we can make sure the Listings edits are implemented correctly.

Thanks all for your input on this! And Happy 2022! 🎉
